src/Documentation/Getting-Started-With-Orleans/Developing-a-Client.md

```diff
@@ -3,71 +3,87 @@ layout: page
+In a typical setup, a frontend web server:
+* Receives a request over HTTP (or some other protocol)
+* Performs necessary authentication and authorization validation
+* Decides which grain(s) should process the request
+* Uses Grain Client to make one or more method call to the grain(s)
+* Handles successful completion or failures of the grain calls and any returned values
+* Sends a response for the HTTP request
 
-Almost any client will involve use of the grain factory class.
-The `GetGrain()` method is used for getting a grain reference for a particular ID.
-As was already mentioned, grains cannot be explicitly created or deleted.
+### Initialization of Grain Client
 
-``` csharp
-GrainClient.Initialize();
+Before a grain client can be used for making calls to grains hosted in an Orleans cluster, it needs to be configured, initialized, and connected to the cluster.
 
-// Hardcoded player ID
-Guid playerId = new Guid("{2349992C-860A-4EDA-9590-000000000006}");
-IPlayerGrain player = GrainClient.GrainFactory.GetGrain<IPlayerGrain>(playerId);
+Configuration is provided via a `ClientConfiguration` object that contains a hierarchy of configuration properties for programmatically configuring a client.
+There is also a way to configure a client via a XML file, but that option will be deprecated in the future.
+More information is in the [Client Configuration guide](..\Orleans-Configuration-Guide\Client-Configuration.md).
+Here we will simply use a helper method that creates a configuration object hardcoded for connecting to a local silo running as `localhost`.
+```csharp
+ClientConfiguration clientConfig = ClientConfiguration.LocalhostSilo(); 
+```
 
-IGameGrain game = player.CurrentGame.Result;
-var watcher = new GameObserver();
-var observer = GrainClient.GrainFactory.CreateObjectReference<IGameObserver>(watcher);
-await game.SubscribeForGameUpdates();
+Once we have a configuration object, we can build a client via the `ClientBuilder` class.
+```csharp
+IClusterClient client = new ClientBuilder().UseConfiguration(clientConfig).Build();
 ```
 
-If this code is used from the main thread of a console application, you have to call `Wait()` on the task returned by `game.SubscribeForGameUpdates()` because `await` does not prevent the `Main()` function from returning, which will cause the client process to exit.
+Lastly, we need to call `Connect()` method on the constructed client object to make it connect to the Orleans cluster, and wait for connection to succeed.
+If the code runs withing an asynchronous method, then it is better to `await` the `Connect()` call, so that we do not block the executing thread:
+```csharp
+await client.Connect(); 
+```
 
-See the Key Concepts section for more details on the various ways to use `Task`s for execution scheduling and exception flow.
+Otherwise, if the running method cannot be made asynchronous, we have to block the thread with a `Wait()`:
+```csharp
+client.Connect().Wait(); 
```

Comment:
Not sure if we need to give the async Tasks 101 here - could it be assumed knowledge / a link? I suppose it's more newbie friendly this way